### ISSUE TITLE
[0.36.0+] When minSpareCols is set to 1, using Tab key to create new columns deletes data from previous columns

### ISSUE THREAD

**Issue by paolostyle** [handsontable/handsontable#4986]
I can't quite explain it, hence the rather bad title, so steps to reproduce:

1. https://jsfiddle.net/ws4acu4n/ - `minSpareCols: 1` is the most relevant option
2. Go to any cell in F column, say F1
3. Type anything, press Tab
4. You'll be in G1, do the same thing - type anything and press Tab
5. You'll be in H1, do the same thing again - type whatever and press Tab
6. The G1 cell is now empty when, obviously, it should not be. 

If you do that a few more times (type anything and press tab to create new column) you'll notice only the last typed value is being displayed. It does not happen when you use arrow keys but it's irrelevant - it should work with Tab too, considering there are no issues with adding new rows with Enter when `minSpareRows` is set to 1. I guess blocking Tab key for users or using `minSpareCols: 2`, in which case, interestingly, there's no such problem, is some kind of workaround, but regardless, it should be fixed.

Handsontable version 0.38.1, bug occurs in at least latest Firefox, Chrome, Edge and IE11.

**Comment by AMBudnik** [comment]
Thank you for sharing the issue @paolostyle 
I have replicated it.

Here is what I've got

![gif](https://user-images.githubusercontent.com/10757813/38240085-d8cfecde-372e-11e8-8dda-fcb1f0403979.gif)

What is interesting is the fact that the data is saved properly when we use <kbd> > </kbd> (right arrow) instead.

Gladly it doesn't happen for `minSpareRows` even if we use <kbd>Tab</kbd> to move down while using `tabMoves: {row: 1, col: 0}` settings.

The bug has been created in the [`0.36.0`/ PRO `1.16.0`](https://docs.handsontable.com/pro/1.16.0/tutorial-release-notes.html)version when we have made the selection changes. It looks like `tabMoves` function uses `selection` start coordinates that are no longer available by the same reference.
